diff --git a/CMakeLists.txt b/CMakeLists.txt
index e1298ba..ca6efba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,10 @@ if(NOT DEFINED QT_MAJOR_VERSION)
 message(FATAL_ERROR "Qt not found")
 endif()
 
+if (Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    find_package(Qt6GuiPrivate REQUIRED)
+endif()
+
 if(NOT USE_PLASMAPKG)
   plasma_install_package(plugin com.github.catsout.wallpaperEngineKde wallpapers
                          wallpaper)
